database/tiny/controller.js

```diff
@@ -184,27 +184,56 @@ class DatabaseController {
 
     /**
      * Use the TinyPEN overwrite endpoint to overwrite the supplied JSON object.
+     * Implements optimistic locking using If-Overwritten-Version header.
      * TODO Pass forward the user bearer token from the Interfaced to TinyPEN?
      * @return the updated JSON or Error
      */
     async overwrite(data) {
         err_out._dbaction = this.URLS.OVERWRITE
+        
+        const headers = {
+            'Content-Type': 'application/json; charset=utf-8'
+        }
+        
+        // Add optimistic locking header if __rerum.isOverwritten exists
+        if (data.__rerum?.isOverwritten) {
```

Comment:
It does seem like requests that come in here always have the __rerum property so this will essentially make every overwrite request out of here have this header.